packages/client/hmi-client/src/views/Model.vue

```diff
@@ -69,55 +55,43 @@ class ModelPlanRenderer extends BaseComputionGraph<NodeData, EdgeData> {
 	}
 }
 
-onMounted(async () => {
-	let renderer: ModelPlanRenderer | null = null;
-	const modelDrawnElement = document.getElementById('model-panel') as HTMLDivElement;
-	const g: IGraph<NodeData, EdgeData> = parsePetriNet2IGraph(modelPetriNet); // get graph from petri net representation
-
-	renderer = new ModelPlanRenderer({
-		el: modelDrawnElement,
-		useAStarRouting: true,
-		runLayout: runDagreLayout
-	});
-
-	// Test on click
-	renderer.on(
-		'node-click',
-		(_eventName: string | symbol, _event, selection: D3SelectionINode<NodeData>) => {
-			console.log(selection.datum());
-		}
-	);
-
-	// write json to model-json and draw model to model-drawn:
-	await renderer?.setData(g);
-	await renderer?.render();
-});
-
 const getModel = async (modelId: string) => API.get(`/models/${modelId}`);
 
-// TODO: let the user choose the model to display
-const selectedModelId = ref('1');
-
 const model = ref<Model | null>(null);
 // Whenever selectedModelId changes, fetch model with that ID
 watch(
-	() => [selectedModelId.value],
+	() => [props.modelId],
 	async () => {
-		const result = await getModel(selectedModelId.value);
+		const result = await getModel(props.modelId);
 		model.value = result.data as Model;
 	},
 	{ immediate: true }
 );
+
+const graphElement = ref<HTMLDivElement | null>(null);
+// Render graph whenever a new model is fetched or whenever the HTML element
+//	that we render the graph to changes.
+watch([model, graphElement], async () => {
+	if (model.value === null || graphElement.value === null) return;
```

Comment:
I find it a little weird to watch a `div` - can we just watch the model but retain the same conditional check?

Reply:
This is to handle the case where model.value is set before the graphElement is created on the page.

In that case we get

1) model is set
2) conditional is true, return early
3) graph element is created
4) graph never gets rendered since we're not watching the graph element

The same thing happens if the graph element is rerendered for any reason.

Both of these possibilities are unlikely but I think it's nice to be extra robust.

Is there a downside I'm missing? If not I'd recommend we'd leave it.

Reply:
Discussed externally.